tests/console/aide_check.pm

```diff
@@ -1,42 +1,73 @@
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: FIPS case for AIDE (Advanced Intrusion Detection Environment) check tool
+#          Test basic function of AIDE and check differences between aide.db and file system
+#
+#          1. Install aide if it has not been installed
+#          2. Initialized the aide database and check
+#          3. Check the difference between datebase and file system
+#          4. Modified the file system and run aide check again
 #
-#    This is a new test case added for FIPS, it test the basic function of aide tools.
-#    This case will been used with SLE 12 SP2 with FIPS, and it has added to misc part of FIPS test.
-#    The case will do operation as followed
-#    1. Install aide if it has not been installed.
-#    2. Initialized the aide database and check
-#    3. Check the difference between datebase and file system.
-#    4. Modified the file system and run aide check again.
-# Maintainer: Jiawei Sun <Jiawei.sun@suse.com>
+# Maintainer: Ben Chou <bchou@suse.com>
+# Tags: poo#64364, tc#1744128
 
 use base "consoletest";
 use testapi;
-use utils;
+use utils "zypper_call";
 use strict;
 use warnings;
 
 sub run {
     select_console 'root-console';
+
     zypper_call "in aide";
+
     assert_script_run "cp /etc/aide.conf /etc/aide.conf.bak";
-    assert_script_run "sed -i 's:^/:!/:g' /etc/aide.conf && sed -i 's:!/var/log:/var/log:g' /etc/aide.conf";
-    assert_script_run "aide -i", 60;
+
+    # Initialize the database and move it to the appropriate place before using the --check command
+    validate_script_output "aide --init || true", sub { m/AIDE initialized database/ }, 300;
+
+    send_key 'ret';
+
     assert_script_run "cp /var/lib/aide/aide.db.new /var/lib/aide/aide.db";
-    assert_script_run "aide --check", 60;
+
+    # Checks the database for inconsistencies and there is 1 new added entry
+    validate_script_output "aide --check || true", sub { m/AIDE found differences between database and filesystem/ && m/Added entries:(\s+)1/ }, 300;
+
+    send_key 'ret';
+
     assert_script_run "touch /var/log/testlog";
-    assert_script_run "clear";
-    validate_script_output "aide --check || true", sub { m/found differences between database and filesystem/ }, 60;
+
+    send_key 'ret';
+
+    # Checks the database for inconsistencies and there is 2 new added entry
+    validate_script_output "aide --check || true", sub { m/AIDE found differences between database and filesystem/ && m/Added entries:(\s+)2/ }, 300;
```

Comment:
why is "||" needed?

Reply:
I think it is the code style question :), I think these 2 methods will both work well.
`"aide --check || true"`  and `"aide --check" ` . It is no harm to keep it.